x/ratelimit/keeper/rate_limit.go

```diff
@@ -19,13 +21,126 @@ func (k Keeper) GetChannelValue(ctx sdk.Context, denom string) sdk.Int {
 	return k.bankKeeper.GetSupply(ctx, denom).Amount
 }
 
-func CheckRateLimit(direction types.PacketDirection, packet exported.PacketI) error {
-	// TODO
-	return nil
+// QUESTION FOR REVIEWER: Where should these two functions live?
```

Comment:
I would prefer to move these two functions into utils/utils.go.
What do you think?

Reply:
I thought about this a bit, and I actually think it might make sense to move ParseDenomFromSendPacket, ParseDenomFromRecvPacket, SendPacket and OnRecvPacket to a separate file called `rate_limit.go` that's at the same level as module.go (since they're not actually keeper functions). I'll take a quick pass at this and push up another commit and we can always revert it if you think it doesn't make sense!

Reply:
@antstalepresh  Just pushed it up, see the latest commits, and let me know what you think!

Reply:
That makes sense!